### PR TITLE
Bugfix fix: "Replace method with property" removes a space

### PR DESCRIPTION
**Pull request by vatsalyaagrawal**
Git Issue: https://github.com/dotnet/roslyn/issues/31830

Issue: Trivia after the bracket was not being considered as part of the rewrite.

Solution: Updated code to consider trivia after the bracket to be the trailing trivia.

Testing:
- Tested with different trivia: spaces, tabs, comments.
- Added new unit test
- Ran all relevant unit tests

### CODE REVIEW

File: src/Features/CSharp/Portable/ReplaceMethodWithProperty/CSharpReplaceMethodWithPropertyService.cs [hunk 1]
```diff
@@ -376,11 +376,12 @@ public void ReplaceSetReference(SyntaxEditor editor, SyntaxToken nameToken, stri
                 return;
             }
 
+            var invocation = nameNode?.FirstAncestorOrSelf<InvocationExpressionSyntax>();
+
             var newName = nameChanged
-                ? SyntaxFactory.IdentifierName(SyntaxFactory.Identifier(propertyName).WithTriviaFrom(nameToken))
+                ? SyntaxFactory.IdentifierName(SyntaxFactory.Identifier(propertyName).WithLeadingTrivia(nameToken.LeadingTrivia).WithTrailingTrivia(invocation?.ArgumentList?.CloseParenToken.TrailingTrivia))
```
Comment (CyrusNajmabadi) [inline src/Features/CSharp/Portable/ReplaceMethodWithProperty/CSharpReplaceMethodWithPropertyService.cs]: if invocation is null, it feels lke we shoudl explicitly handle that, or else we lose the trailing trivia from nameToken.

File: src/EditorFeatures/CSharpTest/CodeActions/ReplaceMethodWithProperty/ReplaceMethodWithPropertyTests.cs [hunk 2]
```diff
@@ -162,6 +162,29 @@ int Goo
...
+        return GetP() == 0;
+    }
+}",
+@"class C
+{
+    int P { get; }
+
+    bool M()
+    {
+        return P == 0;
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/CSharpTest/CodeActions/ReplaceMethodWithProperty/ReplaceMethodWithPropertyTests.cs]: can we have an example where we're not invoking 'GetP' but there is trivia followin it?  i.e. if passing a method in a location where a delegate is needed.  Thanks!

Comment (vatsalyaagrawal) [reply src/EditorFeatures/CSharpTest/CodeActions/ReplaceMethodWithProperty/ReplaceMethodWithPropertyTests.cs]: Added a new test.

File: src/Features/CSharp/Portable/ReplaceMethodWithProperty/CSharpReplaceMethodWithPropertyService.cs (hunk 1 above)
Comment (vatsalyaagrawal) [reply src/Features/CSharp/Portable/ReplaceMethodWithProperty/CSharpReplaceMethodWithPropertyService.cs]: Added doe to handle condition when invocation is null.

File: src/Features/CSharp/Portable/ReplaceMethodWithProperty/CSharpReplaceMethodWithPropertyService.cs [hunk 3]
```diff
@@ -376,11 +376,24 @@ public void ReplaceSetReference(SyntaxEditor editor, SyntaxToken nameToken, stri
...
+                if (invocation == null)
+                {
+                    newName = SyntaxFactory.IdentifierName(SyntaxFactory.Identifier(propertyName)
+                        .WithTriviaFrom(nameToken));
+                }
+                else
+                {
+                    newName = SyntaxFactory.IdentifierName(SyntaxFactory.Identifier(propertyName)
+                        .WithLeadingTrivia(nameToken.LeadingTrivia)
+                        .WithTrailingTrivia(invocation.ArgumentList?.CloseParenToken.TrailingTrivia));
```
Comment (CyrusNajmabadi) [inline src/Features/CSharp/Portable/ReplaceMethodWithProperty/CSharpReplaceMethodWithPropertyService.cs]: i don't believe ArgumentList is optional in a C# invocation list.    Do i don't think you need `?.`

Comment (vatsalyaagrawal) [reply src/Features/CSharp/Portable/ReplaceMethodWithProperty/CSharpReplaceMethodWithPropertyService.cs]: Done.

**Pull request merged**